server/src/main/java/org/apache/cassandra/sidecar/tasks/CassandraClusterSchemaMonitor.java

```diff
@@ -0,0 +1,233 @@
+    }
+
+    @Override
+    public DurationSpec delay()
+    {
+        return sidecarConfiguration.serviceConfiguration().cdcConfiguration().tableSchemaRefreshTime();
+    }
+
+    @Override
+    public void execute(Promise<Void> promise)
+    {
+        try
+        {
+            refresh();
+            promise.tryComplete();
+        }
+        catch (Throwable t)
+        {
+            promise.fail(t);
+        }
+    }
+
+    @Override
+    public ScheduleDecision scheduleDecision()
+    {
+        if (sidecarConfiguration.serviceConfiguration().schemaKeyspaceConfiguration().isEnabled() &&
+            sidecarConfiguration.serviceConfiguration().cdcConfiguration().isEnabled())
+        {
+            return ScheduleDecision.EXECUTE;
+        }
+        return ScheduleDecision.SKIP;
+    }
+
+    @VisibleForTesting
+    static Set<CqlTable> buildCdcTables(CdcDatabaseAccessor cdcDatabaseAccessor,
+                                        ConcurrentHashMap<TableIdentifier, UUID> tableIdCache,
+                                        @NotNull InstanceMetadataFetcher instanceFetcher,
+                                        @NotNull final CassandraBridge cassandraBridge)
+    {
+        return buildCdcTables(cdcDatabaseAccessor.fullSchema(),
+                              cdcDatabaseAccessor.partitioner(),
+                              tableIdCache,
+                              cdcDatabaseAccessor::getTableId,
+                              instanceFetcher,
+                              cassandraBridge);
+    }
+
+    private static Set<CqlTable> buildCdcTables(@NotNull String fullSchema,
+                                                @NotNull CdcDatabaseAccessor cdcDatabaseAccessor,
+                                                @NotNull ConcurrentHashMap<TableIdentifier, UUID> tableIdCache,
+                                                @NotNull InstanceMetadataFetcher instanceFetcher,
+                                                @NotNull final CassandraBridge cassandraBridge)
+    {
+        return buildCdcTables(fullSchema,
+                              cdcDatabaseAccessor.partitioner(),
+                              tableIdCache,
+                              cdcDatabaseAccessor::getTableId,
+                              instanceFetcher,
+                              cassandraBridge);
+    }
+
+    private static Set<CqlTable> buildCdcTables(@NotNull final String fullSchema,
+                                                @NotNull final Partitioner partitioner,
+                                                @NotNull final ConcurrentHashMap<TableIdentifier, UUID> tableIdCache,
+                                                @NotNull final Function<TableIdentifier, UUID> tableIdLoaderFunction,
+                                                @NotNull final InstanceMetadataFetcher instanceFetcher,
+                                                @NotNull final CassandraBridge cassandraBridge)
```

Comment:
we can omit final, and instanceFetcher is no longer needed
```suggestion
    private static Set<CqlTable> buildCdcTables(@NotNull String fullSchema,
                                                @NotNull Partitioner partitioner,
                                                @NotNull ConcurrentHashMap<TableIdentifier, UUID> tableIdCache,
                                                @NotNull Function<TableIdentifier, UUID> tableIdLoaderFunction,
                                                @NotNull CassandraBridge cassandraBridge)
```